app/contexts/AppStateContext.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { createContext, useCallback, useContext, useEffect, useRef } from 'react'
+import { AppState, AppStateStatus } from 'react-native'
+
+export type SimplifiedAppStateStatus = 'active' | 'background'
+export type AppStateListener = (status: SimplifiedAppStateStatus) => void
+
+export interface AppStateContext {
+  addListener: (listener: AppStateListener) => number
+  removeListener: (id: number) => void
+}
+
+const appStateContext = createContext<AppStateContext>(undefined as any)
+
+export function useAppStateContext (): AppStateContext {
+  return useContext(appStateContext)
+}
+
+export function AppStateContextProvider (props: React.PropsWithChildren<any>): JSX.Element | null {
+  // DO NOT use `useState`, these are meant for appstate monitoring
+  // re-render side effect is undesireable
+  const listenerRef = useRef<{ [key: number]: AppStateListener } | null>(null)
+  const prevState = useRef<AppStateStatus>('active') // first mounted
+
+  // all useCallback / useEffect in this context provider MUST have ZERO dependencies, to remain predictable
+  const emit = useCallback((listeners: AppStateListener[], nextState: SimplifiedAppStateStatus) => {
+    prevState.current = nextState // simplify iOS lifecyle, skipping `inactive` as transition state
+    listeners.forEach(l => l(nextState))
+  }, [])
```

Comment:
No point using `useCallback` if no dependencies were passed in, maybe just use a function? 

> Easier to understand, I was trying to understand why we needed a callback.

Reply:
erm, not sure if I misunderstand some react lifecycle. `useCallback` with no dependency so that it will not get reinstantiate on re-render.